experimental/devices/rainbowhat/rainbowhat.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package rainbowhat
+
+import (
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/devices/apa102"
+	"periph.io/x/periph/devices/bmxx80"
+	"periph.io/x/periph/experimental/devices/ht16k33"
+	"periph.io/x/periph/host/rpi"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c/i2creg"
+)
+
+const (
+	numPixels  = 7
+	bmp280Addr = 0x77
+)
+
+// Dev represents a Rainbow HAT  (https://shop.pimoroni.com/products/rainbow-hat-for-android-things)
+type Dev struct {
+	ledstrip *apa102.Dev
+	bmp280   *bmxx80.Dev
+	display  *ht16k33.Display
+	buttonA  gpio.PinIn
+	buttonB  gpio.PinIn
+	buttonC  gpio.PinIn
+	ledR     gpio.PinOut
+	ledG     gpio.PinOut
+	ledB     gpio.PinOut
+	buzzer   gpio.PinOut
+	servo    gpio.PinOut
+}
+
+// NewRainbowHat returns a rainbowhat driver.
+func NewRainbowHat() (*Dev, error) {
+	i2cPort, err := i2creg.Open("")
```

Comment:
Please still have the client pass the I²C and SPI handles explicitly. While we can assume it's fine, there's a risk of using I²C bus # 0 instead of # 1 when it is manually enabled, which is unusual but may happen.

Or

Explicitly request the exact bus, i.e. i1c 1 and spi 0. You can request `/dev/i2c-1` and `/dev/spidev0.0` as printed with spi-list and i2c-list. That said, if an eventual RPi4 would rename the port numbers, this would break this code.